File: checkPermutation.py
#this is in python solution that I have done for better understanding

def is_permutation(s: str, t:str)->bool:
    #1. optimisation: if the strings have different lengths it is impossible to be a permutation 

    if len(s) != len(t):
        return False
    
    #2. sort both strings. if they are permutation, 

#their sorted version will be identical and it says they are permutation

    return sorted(s)==sorted(t)

# Test cases 
if __name__ == "__main__": 
    test_pairs = [
        ("apple", "papel"),   # True
        ("carrot", "tarroc"), # True
        ("hello", "llloh"),   # False
        ("dog", "God"),       # False (case-sensitive)
        (" a bc ", "bca  ") 


    ]

    for s1, s2 in test_pairs:
        result = is_permutation(s1,s2)
        print(f'"{s1}","{s2}" : {result}')

File: checkPermutation.java


import java.util.Arrays;

public class checkPermutation {


    /* a helper method
     * that sorts the character of a string. 
     * @param s the input String.
     * @return a new string with the characters of the input str
     * 
     */
    private static String sortString(String s){
        ///converts the String to a character Array
        /// 
        char[] content = s.toCharArray();
        //
        Arrays.sort(content);
        return new String(content);
    }
    /**
     * Checks if string t is a permutation of string s.
     * This comparison is case-sensitive and whitespace is significant.
     * @param s The first string.
     * @param t The second string.
     * @return true if t is a permutation of s, false otherwise.
     */
    public static boolean isPermutation(String s, String t){
        if (s.length() != t.length()){
            return false;
        }
        // sort both Strings and check if they are equal 
        //if they are permutations, they will become identical once sorted
        return sortString(s).equals(sortString(t));
    }
   

    
    public static void main(String[] args) {
        String[][] pairs = {
            {"apple", "papel"},   {"carrot", "tarroc"}, 
            {"hello", "llloh"},
            {"dog", "God"}, // Case-sensitive
            {" a bc ", "bca  "} // Whitespace-sensitive
        };


        for(String[] pair : pairs){
            String word1 = pair[0];
            String word2 = pair[1];
            boolean result = isPermutation(word1, word2);
            System.out.println("\"" + word1 + "\", \"" + word2 + "\": " + result);

        }
        
    }
}
